full/src/main/java/apoc/dv/Virtualize.java

```diff
@@ -0,0 +1,92 @@
+                .map(VirtualizedResource::toDTO);
+    }
+
+    @Procedure(name = "apoc.dv.catalog.remove", mode = Mode.WRITE)
+    @Description("Remove a virtualized resource config by name")
+    public Stream<VirtualizedResource.VirtualizedResourceDTO> remove(@Name("name") String name) {
+        return new VRCatalogHandler(db, apocConfig.getSystemDb(), log)
+                .remove(name)
+                .map(VirtualizedResource::toDTO);
+    }
+
+    @Procedure(name = "apoc.dv.catalog.list", mode = Mode.READ)
+    @Description("List all virtualized resource configs")
+    public Stream<VirtualizedResource.VirtualizedResourceDTO> list() {
+        return new VRCatalogHandler(db, apocConfig.getSystemDb(), log).list()
+                .map(VirtualizedResource::toDTO);
+    }
+
+    @Procedure(name = "apoc.dv.query", mode = Mode.READ)
+    @Description("Query a virtualized resource by name and return virtual nodes")
+    public Stream<NodeResult> query(@Name("name") String name,
+                                    @Name(value = "params", defaultValue = "{}") Object params,
+                                    @Name(value = "config", defaultValue = "{}") Map<String, Object> config) {
+        VirtualizedResource vr = new VRCatalogHandler(db, apocConfig.getSystemDb(), log).get(name);
+        final Pair<String, Map<String, Object>> procedureCallWithParams = vr.getProcedureCallWithParams(params, config);
+        return tx.execute(procedureCallWithParams.first(), procedureCallWithParams.other())
+                .stream()
+                .map(m -> (Node) m.get(("node")))
+                .map(NodeResult::new);
+    }
+
+    @Procedure(name = "apoc.dv.queryAndLink", mode = Mode.READ)
+    @Description("Query a virtualized resource by name and return virtual nodes linked using virtual rels to the node passed as first param")
+    public Stream<PathResult> queryAndLink(@Name("node") Node node,
+                                           @Name("relName") String relName,
+                                           @Name("name") String name,
+                                           @Name(value = "params", defaultValue = "{}") Object params,
+                                           @Name(value = "config", defaultValue = "{}") Map<String, Object> config) {
+        VirtualizedResource vr = new VRCatalogHandler(db, apocConfig.getSystemDb(), null).get(name);
+        final RelationshipType relationshipType = RelationshipType.withName(relName);
+        final Pair<String, Map<String, Object>> procedureCallWithParams = vr.getProcedureCallWithParams(params, config);
+        return tx.execute(procedureCallWithParams.first(), procedureCallWithParams.other())
+                .stream()
+                .map(m -> (Node) m.get(("node")))
+                .map(n -> new VirtualRelationship(node, n, relationshipType))
+                .map(r -> new PathImpl.Builder(r.getStartNode()).push(r).build())
```

Comment:
there is a `VirtualPath` class you can use instead of this

Reply:
Actually I asked you in another PR to remove `VirtualPath` and use `PathImpl` instead in order to don't manage custom business rules to make the path consistent:
https://github.com/neo4j-contrib/neo4j-apoc-procedures/pull/2157#issuecomment-916794094

Reply:
What's the decision on this? `VirtualPath` could be used here and in #2111.

Reply:
Done, I also added a small builder